### PR TITLE
chore(local): add quality controls

### PR DESCRIPTION
**Pull request by anis-campos**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+
```
Comment (noirbee) [inline .pre-commit-config.yaml:2]: Stray newline

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+repos:
+  - repo: local
+    hooks:
+      - id: yamllint
```
Comment (noirbee) [inline .pre-commit-config.yaml:6]: Use https://github.com/adrienverge/yamllint and pass `args:`

File: local/make/dev.mk [hunk 3]
```diff
@@ -0,0 +1,46 @@
+.PHONY: init sync upgrade_package upgrade_all_requirements upgrade_requirements dev_http_server dev_sync_server dev_importer
```
Comment (noirbee) [inline local/make/dev.mk:1]: Copy&paste error

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Use a standard bash shell, avoid zsh or fish
+SHELL:=/bin/bash
+
+# Select the default target, when you are simply running "make"
+.DEFAULT_GOAL:=init
+
+# Include all the targets
+include local/make/dev.mk
+include local/make/lint.mk
```
Comment (noirbee) [inline Makefile:9]: Mega overkill.

Just create venv + `pip install pre-commit`, no need for `pip-tools`, separate requirements file etc.

**Review by noirbee — CHANGES_REQUESTED** [review]

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+SHELL:=/bin/bash
+
+# Select the default target, when you are simply running "make"
+.DEFAULT_GOAL:=lint
+
+# local executables
+pip:=./venv/bin/pip
+pre-commit=./venv/bin/pre-commit
+
+.PHONY:lint venv
```
Comment (noirbee) [inline Makefile:11]: Missing space after colon

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+venv:
+	python3.7 -m venv venv
+	$(pip) install pre-commit
+
+lint:
+	source venv/bin/activate && \
+	pre-commit run --all-files
```
Comment (noirbee) [inline Makefile:19]: Keep it consistent and use `$(pre-commit) run --all-files`

**Review by noirbee — CHANGES_REQUESTED** [review]
LGTM otherwise

File: Makefile (hunk 6 above)
Comment (anis-campos) [reply Makefile:19]: It was the intention, arf I should go home at that point !

**Review by noirbee — APPROVED** [review]

**Pull request merged**